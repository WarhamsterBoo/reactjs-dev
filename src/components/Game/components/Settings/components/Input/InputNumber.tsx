import React from "react";
import { StyledNumberInput } from "./Input.styled";

export const InputNumber: React.FC<React.HTMLProps<HTMLInputElement>> = (
  props
) => {
  return <StyledNumberInput min="1" max="100" {...props} />;
};

InputNumber.displayName = "InputNumber";
