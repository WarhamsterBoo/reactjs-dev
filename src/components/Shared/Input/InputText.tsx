import React from "react";
import { StyledTextInput } from "./Input.styled";

export const InputText: React.FC<React.HTMLProps<HTMLInputElement>> = (
  props
) => {
  return <StyledTextInput type="text" {...props} />;
};

InputText.displayName = "InputText";
