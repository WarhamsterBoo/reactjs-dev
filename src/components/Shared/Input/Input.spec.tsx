import React from "react";
import renderer from "react-test-renderer";
import { InputNumber } from "./InputNumber";
import { InputText } from "./InputText";

describe("Input Number", () => {
  it("should render", () => {
    const sut = <InputNumber />;

    expect(renderer.create(sut).toJSON()).toMatchSnapshot();
  });
});

describe("Input Text", () => {
  it("should render", () => {
    const sut = <InputText />;

    expect(renderer.create(sut).toJSON()).toMatchSnapshot();
  });
});
