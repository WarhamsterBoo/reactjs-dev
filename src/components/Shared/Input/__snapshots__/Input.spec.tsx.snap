// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Number should render 1`] = `
.emotion-0 {
  border-radius: 4px;
  width: 50px;
  margin-bottom: 5px;
  margin-left: 5px;
  margin-right: 25px;
}

<input
  className="emotion-0"
  max="100"
  min="1"
  type="number"
/>
`;
